src/tables/math.rs

```diff
@@ -0,0 +1,1062 @@
+const UNDERBAR_RULE_THICKNESS_OFFSET: usize = 180;
+const UNDERBAR_EXTRA_DESCENDER_OFFSET: usize = 184;
+const RADICAL_VERTICAL_GAP_OFFSET: usize = 188;
+const RADICAL_DISPLAY_STYLE_VERTICAL_GAP_OFFSET: usize = 192;
+const RADICAL_RULE_THICKNESS_OFFSET: usize = 196;
+const RADICAL_EXTRA_ASCENDER_OFFSET: usize = 200;
+const RADICAL_KERN_BEFORE_DEGREE_OFFSET: usize = 204;
+const RADICAL_KERN_AFTER_DEGREE_OFFSET: usize = 208;
+const RADICAL_DEGREE_BOTTOM_RAISE_PERCENT_OFFSET: usize = 212;
+
+impl<'a> MathConstants<'a> {
+    /// Percentage of scaling down for level 1 superscripts and subscripts. Suggested value: 80%.
+    #[inline]
+    pub fn script_percent_scale_down(&self) -> i16 {
+        Stream::read_at(self.data, SCRIPT_PERCENT_SCALE_DOWN_OFFSET).unwrap_or(0)
+    }
+
+    /// Percentage of scaling down for level 2 (scriptScript) superscripts and subscripts.
+    /// Suggested value: 60%.
+    #[inline]
+    pub fn script_script_percent_scale_down(&self) -> i16 {
+        Stream::read_at(self.data, SCRIPT_SCRIPT_PERCENT_SCALE_DOWN_OFFSET).unwrap_or(0)
+    }
+
+    /// Minimum height required for a delimited expression (contained within parentheses, etc.) to
+    /// be treated as a sub-formula. Suggested value: normal line height × 1.5.
+    #[inline]
+    pub fn delimited_sub_formula_min_height(&self) -> u16 {
+        Stream::read_at(self.data, DELIMITED_SUB_FORMULA_MIN_HEIGHT_OFFSET).unwrap_or(0)
+    }
+
+    /// Minimum height of n-ary operators (such as integral and summation) for formulas in display
+    /// mode (that is, appearing as standalone page elements, not embedded inline within text).
+    #[inline]
+    pub fn display_operator_min_height(&self) -> u16 {
+        Stream::read_at(self.data, DISPLAY_OPERATOR_MIN_HEIGHT_OFFSET).unwrap_or(0)
+    }
+
+    /// White space to be left between math formulas to ensure proper line spacing. For example,
+    /// for applications that treat line gap as a part of line ascender, formulas with ink going
+    /// above (os2.sTypoAscender + os2.sTypoLineGap - MathLeading) or with ink going below
+    /// os2.sTypoDescender will result in increasing line height.
+    #[inline]
+    pub fn math_leading(&self) -> MathValueRecord {
+        self.data.get(MATH_LEADING_OFFSET..)
+            .and_then(|data| MathValueRecord::parse(data, self.data))
+            .unwrap_or_default()
+    }
+
+    /// Axis height of the font.
+    ///
+    /// In math typesetting, the term axis refers to a horizontal reference line used for
+    /// positioning elements in a formula. The math axis is similar to but distinct from the
+    /// baseline for regular text layout. For example, in a simple equation, a minus symbol or
+    /// fraction rule would be on the axis, but a string for a variable name would be set on a
+    /// baseline that is offset from the axis. The `axis_height` value determines the amount of
+    /// that offset.
+    #[inline]
+    pub fn axis_height(&self) -> MathValueRecord {
```

Comment:
Maybe we should return `MathValueRecord?` or does `MathValueRecord` has a defined default value? Or is just to simplify the API?

Reply:
These fields are all required in the spec, and parsing of these fields (including the `Device` table) is delayed until accessed by these methods. Here is the current strategy: either NULL offset for these tables or failure of parsing these `Device` tables gives `None` for `MathValueRecord::device_corrections` fields. `MathValueRecord::parse` can only fail if the `&[u8]` byte slice ends prematurely. I mimicked the behaviour I found here in OS/2 parsing:
https://github.com/RazrFalcon/ttf-parser/blob/fe7984632d9407eef75bf568f77eac1493f8045a/src/tables/os2.rs#L236-L239

Reply:
In practice, I don't expect any of these `unwrap_or_default` get called. Also, I cannot think of any use case of distinguishing between "the `Device` table does not exist in the first place" and "we failed parsing the `Device` table", since either way the `device_corrections` data is just unavailable. I really hope there is a better design, but I have not come up with one yet.

Reply:
Got it. Makes sense.